### DISCUSSION TITLE
IP address from request

### DISCUSSION THREAD

**Discussion by akdavidsson** [actix/actix-web#1897] [Q&A] [answered]
How can I get a client's real IP address from a request?
`#[post("/register")]
pub async fn register(req: HttpRequest, reg: web::Json<Registration>, regs: web::Data<Registrations>) -> Result<String> {
    println!("Got thing: {:?}", req); //No information about IP
    .......
}`

Thanks.

**Comment by fakeshadow** [comment, marked as answer]
You can try `HttpRequest::peer_addr`. And some extra info can be find in [doc](https://docs.rs/actix-web/3.3.2/actix_web/web/struct.HttpRequest.html#method.peer_addr).

**Reply by akdavidsson** [reply]
Thanks for the help!
